contracts/finance/VestingWalletWithCliff.sol

```diff
@@ -0,0 +1,46 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.20;
+
+import {SafeCast} from "../utils/math/SafeCast.sol";
+import {VestingWallet} from "./VestingWallet.sol";
+
+/**
+ * @dev Extension of {VestingWallet} that adds a cliff to the vesting schedule.
+ */
+abstract contract VestingWalletWithCliff is VestingWallet {
+    using SafeCast for *;
+
+    uint64 private immutable _cliff;
+
+    error InvalidCliffDuration(uint64 cliffSeconds, uint64 durationSeconds);
+
+    /**
+     * @dev Sets the sender as the initial owner, the beneficiary as the pending owner, the start timestamp, the
+     * vesting duration and the duration of the cliff of the vesting wallet.
+     */
+    constructor(uint64 cliffSeconds) {
+        if (cliffSeconds > duration()) {
```

Comment:
I don't think this makes a practical difference but imo it makes sense to exclude the case where `cliffSecodns == duration()`

```suggestion
        if (cliffSeconds >= duration()) {
```

Reply:
IMO, if both are equal, that is ok. No real reason to reject.

There is a usecase for the "basic" VestingWallet where start = end (duration = 0) → that release everything at once at some point in the future.

If someone creates a factory for `VestingWalletCliff`, and someone what to deploy an instance that release everything at one, then we would have to do start = cliff = end (duration = 0, cliffSeconds = 0) ... and I think that is ok.